test/gitStash.spec.js

```diff
@@ -0,0 +1,430 @@
+      expect(await gitflow.execGit(['diff', '--cached'], gitOpts)).toEqual(newIndex)
+
+      // Restoring state
+      await gitflow.gitStashPop(gitOpts)
+
+      // Expect stashed files to be back
+      expect(await gitStatus()).toContain(' M test.css')
+      expect(await gitStatus()).toContain('M  test.js')
+      // and content keeps linter modifications
+      expect(await readFile('test.js')).toEqual(newContent)
+      // Expect modifications in index
+      expect(await gitflow.execGit(['diff', '--cached'], gitOpts)).toEqual(newIndex)
+
+      // No stashed should left
+      expect(await gitStashList()).toEqual('')
+    })
+
+    it('should add hooks fixes to index and leave user modifications in WC', async () => {
+      expect(await gitStatus()).toContain(' M test.css')
+      expect(await gitStatus()).toContain(' M test.js')
+
+      // Add test.js to index
+      await gitflow.execGit(['add', 'test.js'], gitOpts)
+      // Save diff for the reference
+      const initialIndex = await gitflow.execGit(['diff', '--cached'], gitOpts)
+
+      // User does additional edits
+      const userContent = `module.exports = {
+    test: 'test2',
+    test: 'test3'
+}`
+      await fsp.writeFile(path.join(wcDirPath, 'test.js'), userContent)
+
+      // Expect test.js is in both index and modified
+      expect(await gitStatus()).toContain(' M test.css')
+      expect(await gitStatus()).toContain('MM test.js')
+      expect(await gitflow.execGit(['diff', '--cached'], gitOpts)).toEqual(initialIndex)
+
+      // Stashing state
+      await gitflow.gitStashSave(gitOpts)
+
+      // Only index should remain
+      expect(await gitStatus()).not.toContain(' M test.css')
+      expect(await gitStatus()).toContain('M  test.js')
+
+      // Do additional edits (imitate eslint --fix)
+      await fsp.writeFile(
+        path.join(wcDirPath, 'test.js'),
+        `module.exports = {
+    test: "test2",
+    test: "test3",
+};`
+      )
+      // and add to index
+      await gitflow.execGit(['add', 'test.js'], gitOpts)
+      const newIndex = await gitflow.execGit(['diff', '--cached'], gitOpts)
+
+      // Expect index is modified
+      expect(await gitflow.execGit(['diff', '--cached'], gitOpts)).not.toEqual(initialIndex)
+      expect(await gitflow.execGit(['diff', '--cached'], gitOpts)).toEqual(newIndex)
+
+      // Restoring state
+      await gitflow.gitStashPop(gitOpts)
+
+      // Expect stashed files to be back
+      expect(await gitStatus()).toContain(' M test.css')
+      expect(await gitStatus()).toContain('MM test.js')
+      // and content is back to user modifications
+      expect(await readFile('test.js')).toEqual(userContent)
+      // Expect modifications in index
+      expect(await gitflow.execGit(['diff', '--cached'], gitOpts)).toEqual(newIndex)
+
+      // No stashed should left
+      expect(await gitStashList()).toEqual('')
+    })
+
+    it('should add hooks fixes to index and working copy on partially staged files', async () => {
```

Comment:
@okonet, could you fix this test? It is failing on CI - https://travis-ci.org/okonet/lint-staged/jobs/334488727#L507

Reply:
I think to write tests in async/await, the correct format is `await expect(fn()).resolves.toEqual(...)`

Reply:
Or is it just a convention?

Reply:
You can use `resolves` if you want, but just using `await` is often cleaner. Depends on your asserition.

```js
await expect(someFunc()).resolves.toBe('cool');
```

vs
```js
const value = await someFunc();
expect(value).toBe('cool');
```

Reply:
@SimenB ah I see, thanks! 👍🏻